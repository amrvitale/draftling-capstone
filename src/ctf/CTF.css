* {
    box-sizing: border-box;
  }

  body {
      margin: 0;
  }

  ul {
    display: inline-block;
    margin: 0;
    padding: 0;
}

  div.critique {
    margin-left: 0;
    margin-right: 0;
    padding: 5px;
    border: 1px solid maroon;
    border-radius: 5px;
    background-color: black;
    font-size: 20px;
    min-width: 300px;

  }

  


.critText {
  font-size: 20px;
}

p {
  white-space: pre-wrap; 
}

