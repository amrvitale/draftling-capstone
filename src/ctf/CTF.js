import React from 'react'
import ApiContext from '../ApiContext'
import './CTF.css'

class CTF extends React.Component {
    static contextType = ApiContext;
    render() {
       const { plot, pov, characters, dialogue, gramspell, overall } = this.props
       
        return (
            <div className="critique">
                <p>{plot}</p>
                <p>{pov}</p>
                <p>{characters}</p>       
                <p>{dialogue}</p>     
                <p>{gramspell}</p>  
                <p>{overall}</p>          
            </div>
        );
       
    }
};

export default CTF;

//if else like with status button, using column as flag