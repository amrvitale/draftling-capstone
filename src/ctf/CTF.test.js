import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CTF from './CTF';

describe (`CTF component`, () => {
    it('renders a .CTF by default', () => {
        const wrapper = shallow(<CTF crit={{plot: "test", pov: "first person", characters: "Angela", dialogue: "hello", gramspell:"none", overall:"good work!"}}/>)
        expect (toJson(wrapper)).toMatchSnapshot()
    })  
})