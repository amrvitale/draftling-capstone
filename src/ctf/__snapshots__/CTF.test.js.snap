// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTF component renders a .CTF by default 1`] = `
<div
  className="critique"
>
  <h3>
    Template Critique
  </h3>
  <p
    className="critText"
  >
    Plot: 
    test
  </p>
   
  <br />
  <p
    className="critText"
  >
    POV: 
    first person
  </p>
   
  <br />
  <p
    className="critText"
  >
    Characters: 
    Angela
  </p>
   
  <br />
  <p
    className="critText"
  >
    Dialogue: 
    hello
  </p>
  <br />
  <p
    className="critText"
  >
    Grammar and spelling: 
    none
  </p>
  <br />
  <p
    className="critText"
  >
    Overall: 
    good work!
  </p>
  <br />
</div>
`;
